nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkStatus.java

```diff
@@ -35,69 +34,76 @@
     private static final long serialVersionUID = 1L;
 
     /**
-     * Possible states of bulk operation.
+     * Possible states of bulk action execution.
      */
     public enum State {
-        SCHEDULED, BUILDING, COMPLETED
+        /** The {@link BulkCommand} has been submitted to the system. */
+        SCHEDULED,
+
+        /** System is currently scrolling the database. */
+        BUILDING,
+
+        /** System has finished to scroll. */
+        COMPLETED
     }
 
-    protected UUID uuid;
+    protected String id;
 
     protected BulkCommand command;
 
     protected State state;
 
-    protected ZonedDateTime creationDate;
+    protected Instant creationInstant;
 
-    protected Long scrolledDocumentCount;
+    protected Long count;
 
     /**
-     * Gets bulk operation id.
+     * Gets bulk action id.
      *
      * @return the id
      */
-    public UUID getUUID() {
-        return uuid;
+    public String getId() {
+        return id;
     }
 
     /**
-     * Sets bulk operation id.
+     * Sets bulk action id.
      *
-     * @param uuid the id
+     * @param id the id
      */
-    public void setUUID(UUID uuid) {
-        this.uuid = uuid;
+    public void setId(String id) {
+        this.id = id;
     }
 
     /**
-     * Gets bulk operation command.
+     * Gets bulk command.
```

Comment:
weird to store the command inside its status

Reply:
Do you think it's useless ? We think it could be useful for UI to be able to retrieve the command in the same call

Reply:
might be easier to have 2 keys in the KV store one for the Command and one for the Status